xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -37,135 +40,129 @@ CEdl::CEdl()
+    if (!found)
+      found = ReadComskip(path, fps);
 
-    if (!bFound)
-      bFound = ReadBeyondTV(strMovie);
+    if (!found)
+      found = ReadBeyondTV(path);
   }
   else
   {
-    bFound = ReadPvr(fileItem);
+    found = ReadPvr(fileItem);
   }
 
-  if (bFound)
+  if (found)
   {
     MergeShortCommBreaks();
     AddSceneMarkersAtStartAndEndOfEdits();
   }
 
-  return bFound;
+  return found;
 }
 
-bool CEdl::ReadEdl(const std::string& strMovie, const float fFramesPerSecond)
+bool CEdl::ReadEdl(const std::string& path, float fps)
 {
   Clear();
 
-  std::string edlFilename(URIUtils::ReplaceExtension(strMovie, ".edl"));
+  std::string edlFilename{URIUtils::ReplaceExtension(path, ".edl")};
   if (!CFile::Exists(edlFilename))
     return false;
 
   CFile edlFile;
   if (!edlFile.Open(edlFilename))
   {
-    CLog::Log(LOGERROR, "{} - Could not open EDL file: {}", __FUNCTION__,
-              CURL::GetRedacted(edlFilename));
+    CLog::LogF(LOGERROR, "Could not open EDL file: {}", CURL::GetRedacted(edlFilename));
     return false;
   }
 
-  bool bError = false;
-  int iLine = 0;
-  std::string strBuffer;
-  strBuffer.resize(1024);
-  while (edlFile.ReadString(&strBuffer[0], 1024))
+  bool error{false};
+  int line{0};
+  std::array<char, 1024> buffer{};
+  while (edlFile.ReadString(&buffer[0], 1024))
   {
     // Log any errors from previous run in the loop
-    if (bError)
-      CLog::Log(LOGWARNING, "{} - Error on line {} in EDL file: {}", __FUNCTION__, iLine,
-                CURL::GetRedacted(edlFilename));
+    if (error)
+      CLog::LogF(LOGWARNING, "Error on line {} in EDL file: {}", line,
+                 CURL::GetRedacted(edlFilename));
 
-    bError = false;
+    error = false;
+    line++;
 
-    iLine++;
+    std::array<char, 513> buffer1{};
+    std::array<char, 513> buffer2{};
+    int action{EDL::EDL_ACTION_NONE};
 
-    char buffer1[513];
-    char buffer2[513];
-    int iAction;
-    int iFieldsRead = sscanf(strBuffer.c_str(), "%512s %512s %i", buffer1,
-                             buffer2, &iAction);
-    if (iFieldsRead != 2 && iFieldsRead != 3) // Make sure we read the right number of fields
+    int fieldsRead =
+        sscanf(buffer.data(), "%512s %512s %i", buffer1.data(), buffer2.data(), &action);
+    // Make sure we read the right number of fields
+    if (fieldsRead != 2 && fieldsRead != 3)
     {
-      bError = true;
+      error = true;
       continue;
     }
 
-    std::vector<std::string> strFields(2);
-    strFields[0] = buffer1;
-    strFields[1] = buffer2;
+    std::array<std::string, 2> fields{};
+    fields[0] = buffer1.data();
+    fields[1] = buffer2.data();
 
-    if (iFieldsRead == 2) // If only 2 fields read, then assume it's a scene marker.
+    // If only 2 fields read, then assume it's a scene marker.
+    if (fieldsRead == 2)
     {
-      iAction = atoi(strFields[1].c_str());
-      strFields[1] = strFields[0];
+      action = atoi(fields[1].c_str());
```

Comment:
```suggestion
      action = std::stoi(fields[1]);
```
ref: https://en.cppreference.com/w/cpp/string/basic_string/stol